src/transformers/modeling_bart.py

```diff
@@ -86,7 +86,7 @@ def _prepare_bart_decoder_inputs(
             causal_lm_mask = None
         new_shape = (bsz, tgt_len, tgt_len)
         # make it broadcastable so can just be added to the attention coefficients
-        decoder_attn_mask = _combine_masks(decoder_padding_mask, causal_lm_mask, new_shape)
+        decoder_attn_mask = _combine_masks(decoder_padding_mask, causal_lm_mask, new_shape).to(device=input_ids.device)
```

Comment:
@sshleifer I have no idea if that's the most correct way to do this, depending on the semantics of the operations around.

Reply:
Unfortunately, `decoder_attn_mask` can be None occasionally.

Reply:
not True, actually.